### DISCUSSION TITLE
Any way to block all WebRTC requests on Electron?

### DISCUSSION THREAD

**Discussion by Cat-Ling** [ghostery/adblocker#3761] [General]
I'm writing an electron webapp that uses random public proxies at startup, but the real public IP always ends up leaking. Electron doesn't have any option to disable it, at least not in their documentation.

Is there any way to block WebRTC everywhere using adblocker?

**Comment by Cat-Ling** [comment]
After days of searching around on every search engine, I found the fix 2 hours after I ask for help.
Anyway, here's help if anybody comes searching for this here since Electron doesn't care to document the code they've themselves added.

1. Create webrtc.js
```js
//webrtc.js
function setWebRTCBlockingPolicy(webContents) {
    webContents.setWebRTCIPHandlingPolicy('disable_non_proxied_udp'); // This will prevent both local and public IP from leaking.
  }
  
  module.exports = { setWebRTCBlockingPolicy };
  
```

2. Import into your script- 
```js
const { setWebRTCBlockingPolicy } = require('./webrtc');
```

3. Add this code before you create your window, this makes sure every sub-window you open also has webRTC disabled.
Add just below *app.whenReady()*
```js
  app.on('browser-window-created', (event, window) => {
    setWebRTCBlockingPolicy(window.webContents);
});
```

We can probably directly do _window.webContents.setWebRTCIPHandlingPolicy('disable_non_proxied_udp');_ instead of importing and calling the function, but I'm just too tired at this point. You probably know what to do.

**Comment by GoCoGit** [comment]
cool, bro. you've done me a big favor
